modules/rpc/rpc.js

```diff
@@ -207,15 +97,23 @@ function init(options) {
 
   // This is a factory function that returns an Observable
   function createObservable(event, method, params) {
-    let auth = getAuth(options);
+    let auth = cookie.getAuth(options);
     return Observable.create(observer => {
-      rpcCall(method, params, auth, (error, response) => {
-        if (error) {
-          observer.error(error);
-          return;
-        }
-        observer.next(response);
-      });
+      if (['restart-daemon'].includes(method)) {
```

Comment:
Sounds like a bad idea to have it like rpc.call('restart-daemon')
If the code is ran in a browser then it won't pass through this code and it will return an error.

I think you're better off adding another channel to the IPC whitelist specifically for this type commands.

Reply:
If it runs in a browser there is no way to restart the daemon. We should handle it appropriately, but the issue isn't here, it's in the encrypt modal where it should be handled.